File: docs/examples/2_register.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
)

func Register() {
	client := &http.Client{}
	payload := map[string]interface{}{
		"email": "newuser@example.com",
		"password": "SecurePass123!",
		"firstName": "Grace",
		"lastName": "Hopper",
		"role": "student"
	}
	body, _ := json.Marshal(payload)
	req, _ := http.NewRequest("POST", "http://localhost:3000/auth/register", bytes.NewBuffer(body))
	req.Header.Add("Content-Type", "application/json")

	resp, err := client.Do(req)
	if err != nil {
		fmt.Println("Error:", err)
		return
	}
	defer resp.Body.Close()

	body, _ := ioutil.ReadAll(resp.Body)
	var result interface{}
	json.Unmarshal(body, &result)
	fmt.Println("Response:", result)
}

func main() {
	register()
}

File: docs/examples/2_register.cs
using System;
using System.Net.Http;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

public class RegisterExample {
    public static async Task Register() {
        using (var client = new HttpClient()) {
            client.BaseAddress = new Uri("http://localhost:3000");
            
            var content = new StringContent(
                JsonSerializer.Serialize({"email":"newuser@example.com","password":"SecurePass123!","firstName":"Grace","lastName":"Hopper","role":"student"}),
                Encoding.UTF8,
                "application/json"
            );

            var response = await client.postAsync("/auth/register", content);
            
            var result = await response.Content.ReadAsStringAsync();
            Console.WriteLine("Response: " + result);
        }
    }

    public static void Main() {
        Register().Wait();
    }
}